canister/src/api/set_config.rs

```diff
@@ -23,6 +23,10 @@ fn set_config_no_verification(request: SetConfigRequest) {
                     .expect("stability threshold too large"),
             );
         }
+
+        if let Some(api_access) = request.api_access {
```

Comment:
Can we add a test case for this? See below for examples on how we test the other configurations.